### PR TITLE
[Plugin] Serve plugin for Denopack

### PR DESCRIPTION
**Pull request by Soremwar**
Works as expected, couple reworks I wanna do before this is landed

### TODO
- [x] Add support for https server
- [x] (If possible) refactor options so they are not a complete mess to handle internally
- [x] Add documentation

To test it, just import and use the plugin. You can pass either a string(s) or an options object to start it

### CODE REVIEW

**Comment by Soremwar** [comment]
@jeroenptrs All set!

File: plugin/serve/mod.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import {
+  createRequestHandler,
+} from "./server.ts";
+
+/** 
+ * Plugin for serving your compiled files through a http/https server
+ * Requires --allow-net=host:port
+ * Can be used in conjuction with --watch for a development server
+ */
+export const serve = (
```
Comment (jeroenptrs) [inline plugin/serve/mod.ts:21]:
```diff
- export const serve = {
+ export const pluginServe = {
```

File: plugin/serve/mod.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    createHTTPServer({
+      hostname: parsed_options.host,
+      port: parsed_options.port,
+    }, createRequestHandler(parsed_options));
+  }
+
+  let running = parsed_options.verbose === false;
+
+  return {
+    name: "serve",
```
Comment (jeroenptrs) [inline plugin/serve/mod.ts:65]:
```diff
- name: "serve"
+ name: "denopack-plugin-serve"
```

**Review by jeroenptrs — CHANGES_REQUESTED** [review]
I'd prefer if the plugin name was a bit more in line with the rest of the plugins. After that we can merge this in, since it's looking really good 😄

**Review by jeroenptrs — APPROVED** [review]
LGTM! 🎉

**Pull request merged**